### PR TITLE
Allow TF Asymmetry Fitting for Diffs between Groups

### LINKED ISSUE #31456: Muon Analysis - Allow TF Asymmetry fitting for the differences between groups
**Original reporter:** Peter Baker

### Description
We think it would be useful to be able to do TF Asymmetry fitting for the difference between groups.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR ensures that TF Asymmetry fitting is allowed for the diff between groups.

**To test:**
1. Open Muon Analysis
2. Load `MUSR62260`
3. Go to Grouping tab and create a Diff between groups called `diff_group`
4. Tick `Analyse (Plot/Fit)`
5. Go to the Fitting tab and select `TF Asymmetry` fitting mode
6. It should switch without a popup warning
7. Create a pair diff and ensure that these are not allowed for TF Asymmetry fitting mode

Fixes #31456

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py [hunk 1]
```diff
@@ -230,11 +230,20 @@ def automatically_update_function_name(self) -> None:
...
         workspace_names = self.fitting_context.dataset_names if workspace_names is None else workspace_names
-        pair_names = [get_group_or_pair_from_name(name) for name in workspace_names if "Group" not in name]
         # Remove duplicates from the list
-        pair_names = list(dict.fromkeys(pair_names))
+        pair_names = list(dict.fromkeys(self._get_pair_and_pair_diff_names(workspace_names)))
         return len(pair_names) == 0, "'" + "', '".join(pair_names) + "'"
 
+    def _get_pair_and_pair_diff_names(self, workspace_names: list) -> list:
+        """Returns the names of pairs and pair diffs which are in the provided workspace name list."""
+        group_diff_names = [diff.name for diff in self.context.group_pair_context.get_diffs("group")]
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py:239]: could we not just do:

```suggestion
        pair_names = [diff.name for diff in self.context.group_pair_context.get_diffs("pair")]
        pair_names += [pair.name for pair in self.context.group_pair_context.pairs]
```

Comment (robertapplin) [reply scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py]: Thanks that is much better

**Comment by AnthonyLim23** [comment]
If you try to do a TF Asymmetry fit on your group diff it throws an error `Python object is None type`

**Comment by AnthonyLim23** [comment]
I have just spoken to Peter. The method used here does not produce an expected result. What they meant to do was to normalise the individual groups first and then take the difference. This is what they meant by TF Asymmetry with group differences (which is the functionality we already have).

**Pull request closed without merging**